src/Symfony/Contracts/HttpClient/ApiClientInterface.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Contracts\HttpClient;
+
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\Test\ApiClientTestCase;
+
+/**
+ * Provides flexible methods for interacting with HTTP APIs.
+ *
+ * Unless otherwise specified, implementations MUST send a JSON-compatible
+ * "accept" header - typically "application/json".
+ *
+ * @see HttpClientInterface for a description of options and how they MUST work
+ * @see ApiClientTestCase   for a reference test suite
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+interface ApiClientInterface
```

Comment:
imho it would be useful to also have the generic `request` method on this interface, for APIs that accept special http methods.

Reply:
We already have such a thing: it's `HttpClientInterface`

Reply:
but you do not extend the HttpClientInterface here. there is a very small cost to also proxy that base method (assuming that the api client will implement such a method anyways)